extensions/functions/src/main/java/io/crate/window/OffsetValueFunctions.java

```diff
@@ -50,64 +52,68 @@
  */
 public class OffsetValueFunctions implements WindowFunction {
 
-    private Object getValueAtOffsetIgnoringNulls(int idxInPartition,
+    private Object getValueAtOffsetIgnoringNulls(LongConsumer nonRowMemoryReserver,
```

Comment:
Can we rename this throughout to just `allocateBytes`?

The "nonRowMemoryReserver" is in a way leaking internals in how exactly it is used und overly verbose.
It is a general contract that components should only account for memory they created (e.g. whenever there is a `materialize` call on a row.
But not on data received from other components.

Reply:
Done in https://github.com/crate/crate/pull/16054/commits/d022f72dcef44606135594f8e45a0f5336c752de